src/main/groovy/wooga/gradle/unity/UnityPluginExtension.groovy

```diff
@@ -256,4 +257,17 @@ trait UnityPluginExtension implements UnitySpec,
         targets
     }
 
+    private final MapProperty<String, String> upmPackages = objects.mapProperty(String, String)
+
+    /**
+     * @return The UPM packages to add
+     */
+    MapProperty<String, String> getUpmPackages() {
+        upmPackages
+    }
+
+    void setUpmPackages(MapProperty<String, String> upmPackages) {
+        upmPackages.clear()
+        upmPackages.putAll(upmPackages)
```

Comment:
If you cann the `set` method on the `MapProperty` it should clear the map before. That is why the `MapProperty` has 2 different API. `put`/`putAll` to append to the map. And `set`/`value`

see https://docs.gradle.org/current/javadoc/org/gradle/api/provider/MapProperty.html